great_expectations/validator/metrics_calculator.py

```diff
@@ -0,0 +1,399 @@
+    from great_expectations.expectations.metrics.metric_provider import MetricProvider
+
+
+MAX_METRIC_COMPUTATION_RETRIES: int = 3
+
+
+class MetricsCalculator:
+    def __init__(
+        self,
+        execution_engine: ExecutionEngine,
+        show_progress_bars: bool = False,
+    ) -> None:
+        """
+        MetricsCalculator accepts and processes metrics calculation requests.
+
+        Args:
+            execution_engine: ExecutionEngine to perform metrics computation.
+        """
+        self._execution_engine: ExecutionEngine = execution_engine
+        self._show_progress_bars: bool = show_progress_bars
+
+    @property
+    def show_progress_bars(self) -> bool:
+        return self._show_progress_bars
+
+    @show_progress_bars.setter
+    def show_progress_bars(self, enable: bool) -> None:
+        self._show_progress_bars = enable
+
+    def columns(self, domain_kwargs: Optional[Dict[str, Any]] = None) -> List[str]:
+        """
+        Convenience method to run "table.columns" metric.
+        """
+        if domain_kwargs is None:
+            domain_kwargs = {
+                "batch_id": self._execution_engine.batch_manager.active_batch_data_id,
+            }
+
+        columns: List[str] = self.get_metric(
+            metric=MetricConfiguration(
+                metric_name="table.columns",
+                metric_domain_kwargs=domain_kwargs,
+            )
+        )
+
+        return columns
+
+    def head(
+        self,
+        n_rows: int = 5,
+        domain_kwargs: Optional[Dict[str, Any]] = None,
+        fetch_all: bool = False,
+    ) -> pd.DataFrame:
+        """
+        Convenience method to run "table.head" metric.
+        """
+        if domain_kwargs is None:
+            domain_kwargs = {
+                "batch_id": self._execution_engine.batch_manager.active_batch_data_id,
+            }
+
+        data: Any = self.get_metric(
+            metric=MetricConfiguration(
+                metric_name="table.head",
+                metric_domain_kwargs=domain_kwargs,
+                metric_value_kwargs={
+                    "n_rows": n_rows,
+                    "fetch_all": fetch_all,
+                },
+            )
+        )
+
+        if isinstance(
+            self._execution_engine, (PandasExecutionEngine, SqlAlchemyExecutionEngine)
+        ):
+            df = pd.DataFrame(data=data)
+        elif isinstance(self._execution_engine, SparkDFExecutionEngine):
+            rows: List[Dict[str, Any]] = [datum.asDict() for datum in data]
+            df = pd.DataFrame(data=rows)
+        else:
+            raise ge_exceptions.GreatExpectationsError(
+                "Unsupported or unknown ExecutionEngine type encountered in Validator class."
+            )
+
+        return df.reset_index(drop=True, inplace=False)
+
+    def get_metric(
+        self,
+        metric: MetricConfiguration,
+    ) -> Any:
+        """return the value of the requested metric."""
+        return self.get_metrics(
```

Comment:
This could call `compute_metrics`. We could rename method to `compute_metric`.

Reply:
@NathanFarmer Not exactly -- the signatures are different, and if we did, we would repeat the transformation already in `get_metrics()`.

Reply:
@alexsherstinsky see my latest comment on line 135. This would mean `get_metric` and `get_metric` go away entirely and would be replaced with `compute_metric` and `compute_metrics`. Anything from the legacy API would call `compute_metric` or `compute_metrics` under the hood.